src/main/java/com/azure/spring/dev/tools/actions/UpdateSpringCloudAzureSupportFileRunner.java

```diff
@@ -115,13 +115,18 @@ private String findCompatibleSpringCloudVersion(String springBootVersion) {
         if (supportMetadata != null && supportMetadata.getSupportStatus() == SupportStatus.END_OF_LIFE) {
             return supportMetadata.getSpringCloudVersion();
         }
-        return this.springCloudCompatibleSpringBootVersionRanges
-            .entrySet()
-            .stream()
-            .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
-            .map(Map.Entry::getKey)
-            .collect(Collectors.toList())
-            .stream().findFirst().orElseThrow();
+        try {
+            return this.springCloudCompatibleSpringBootVersionRanges
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getValue().match(Version.parse(springBootVersion)))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList())
+                .stream().findFirst().orElseThrow();
+        } catch (Exception e) {
+            LOGGER.info("Spring Boot " + springBootVersion + " does not support yet, so skip it");
+        }
+        return "not get the supported Spring Cloud version yet";
```

Comment:
Could you explain this logic, why the error message is like the Spring Boot version is not supported yet?

Reply:
We get a compatible Spring Cloud version from [metadata](https://start.spring.io/actuator/info) for each Spring Boot version, But when a new Spring Boot version was published [here](https://api.spring.io/project_metadata/spring-boot), the supported version may not update yet, for example, Spring Boot version `3.1.0-M2` and `3.1.0-SNAPSHOT` not supported in metadata now:
![image](https://user-images.githubusercontent.com/92105726/232359691-88c4bb68-1ac8-499d-a480-23e614864513.png)